library/js/utility.js

```diff
@@ -575,7 +576,7 @@ if (typeof top.userDebug !== 'undefined' && (top.userDebug === '1' || top.userDe
                 let title = node.dataset.smartName || JSON.stringify(xl("Smart App"));
                 // we allow external dialog's  here because that is what a SMART app is
                 let height = window.top.innerHeight; // do our full height here
-                dlgopen(url, '_blank', 'modal-full', height, '', title, {allowExternal: false, onClose: function() {
+                dlgopen(url, 'smartDsiEditSource', 'modal-full', height, '', title, {allowExternal: false, onClose: function() {
                     window.top.removeEventListener('message', windowMessageHandler);
                 }});
             });
```

Comment:
```
dlgopen(url, 'smartDsiEditSource', 'modal-full', height, '', title, {
                allowExternal: false; //default false so not needed. I need to look into this option!
                resolvePromiseOn: 'close',
                sizeHeight: 'full'
            }).then(function (resolve) {
                top.restoreSession();
                window.top.removeEventListener('message', windowMessageHandler);
            });
```

Preference to resolve via a promise.
unsure why a window name is needed but doesn't hurt in any scope. I'd be curious if window name is removed using promise. I predict it's not needed.

Reply:
thanks Jerry, yep, window name has to be declared so that firefox doesn't mistakenly use 'pat' as the window name.
Chrome isn't able to identify a window name so it's set to an empty string and dlgclose is able to find the [frameName](
https://github.com/openemr/openemr/blob/master/library/dialog.js#L407).